kyuubi-server/src/main/scala/org/apache/kyuubi/operation/KyuubiOperation.scala

```diff
@@ -26,18 +26,18 @@ import org.apache.thrift.TException
 import org.apache.thrift.transport.TTransportException
 
 import org.apache.kyuubi.{KyuubiSQLException, Utils}
-import org.apache.kyuubi.client.KyuubiSyncThriftClient
 import org.apache.kyuubi.metrics.MetricsConstants.STATEMENT_FAIL
 import org.apache.kyuubi.metrics.MetricsSystem
 import org.apache.kyuubi.operation.FetchOrientation.FetchOrientation
 import org.apache.kyuubi.operation.OperationType.OperationType
-import org.apache.kyuubi.session.Session
+import org.apache.kyuubi.session.{KyuubiSessionImpl, Session}
 import org.apache.kyuubi.util.ThriftUtils
 
-abstract class KyuubiOperation(
-    opType: OperationType,
-    session: Session,
-    client: KyuubiSyncThriftClient) extends AbstractOperation(opType, session) {
+abstract class KyuubiOperation(opType: OperationType, session: Session)
+  extends AbstractOperation(opType, session) {
+
+  protected def client = session.asInstanceOf[KyuubiSessionImpl].sessionManager
+    .operationManager.getThriftClient(session.handle)
```

Comment:
move the client here.

As it might be null before the engine session is ready.